ccl/schemachangerccl: TestBackupSuccessMixedVersion_base_alter_table_multiple_commands failed

ccl/schemachangerccl.TestBackupSuccessMixedVersion_base_alter_table_multiple_commands [failed](https://mesolite.cluster.engflow.com/invocations/default/aaa1e247-6350-4654-b3fe-5a5639607f24?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NjaGVtYWNoYW5nZXJjY2w6c2NoZW1hY2hhbmdlcmNjbF90ZXN0) on master @ [3ca67e34d648ba46fb651e8e7667a1715c0427d7](https://github.com/cockroachdb/cockroach/commits/3ca67e34d648ba46fb651e8e7667a1715c0427d7):

Fatal error:

```
panic: test timed out after 15m0s
running tests:
	TestBackupSuccessMixedVersion_base_alter_table_multiple_commands (1m32s)
	TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_16_of_31 (4s)

```
Stack: 

```
goroutine 1486797 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackupSuccessMixedVersion_base_alter_table_multiple_commands3382257137
    test_log_scope.go:76: use -show-logs to present logs inline
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/schemachangerccl_test_/schemachangerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/schemachanger/testdata/end_to_end/alter_table_multiple_commands/alter_table_multiple_commands.definition:1:
        setup [0 args]
        CREATE TABLE t (i INT PRIMARY KEY, j INT NOT NULL, k INT, FAMILY "primary" (i, j, k));
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/schemachangerccl_test_/schemachangerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/schemachanger/testdata/end_to_end/alter_table_multiple_commands/alter_table_multiple_commands.definition:5:
        stage-exec [2 args]
        INSERT INTO t(j,i) VALUES($stageKey, $stageKey);
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/schemachangerccl_test_/schemachangerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/schemachanger/testdata/end_to_end/alter_table_multiple_commands/alter_table_multiple_commands.definition:11:
        stage-query [2 args]
        SELECT count(*)=$successfulStageCount FROM t;
        ----
        true
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_10_of_31
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_11_of_31
    backup.go:105: skipping due to randomness
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_13_of_31/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_1_of_31
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_4_of_31/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_7_of_31/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_7_of_31/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_8_of_31
    backup.go:105: skipping due to randomness
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_13_of_31/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_14_of_31
    backup.go:105: skipping due to randomness
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_15_of_31
    backup.go:105: skipping due to randomness
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_3_of_31
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_3_of_31/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_4_of_31
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_7_of_31
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_10_of_31/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_13_of_31/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_1_of_31/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_3_of_31/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_9_of_31/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_10_of_31/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_10_of_31/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_13_of_31
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_1_of_31/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_5_of_31
    backup.go:105: skipping due to randomness
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_16_of_31
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_base_alter_table_multiple_commands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47986

ccl/schemachangerccl.TestBackupSuccessMixedVersion_base_alter_table_multiple_commands [failed](https://tanzanite.cluster.engflow.com/invocations/default/9e5af20d-05a5-4669-bd47-d5f65e5ec27a?testReportRun=10&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NjaGVtYWNoYW5nZXJjY2w6c2NoZW1hY2hhbmdlcmNjbF90ZXN0) on master @ [e527f6b34c009ed7b52777e854687fb5d6ce3378](https://github.com/cockroachdb/cockroach/commits/e527f6b34c009ed7b52777e854687fb5d6ce3378):

Fatal error:

```
panic: test timed out after 15m0s
	running tests:
		TestBackupSuccessMixedVersion_base_alter_table_multiple_commands (3m26s)
		TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_non_revertible_stage_2_of_4 (4s)

```
Stack: 

```
goroutine 1059795 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2373 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:215 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackupSuccessMixedVersion_base_alter_table_multiple_commands642859139
    test_log_scope.go:76: use -show-logs to present logs inline
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/schemachangerccl_test_/schemachangerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/schemachanger/testdata/end_to_end/alter_table_multiple_commands/alter_table_multiple_commands.definition:1:
        setup [0 args]
        CREATE TABLE t (i INT PRIMARY KEY, j INT NOT NULL, k INT, FAMILY "primary" (i, j));
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/schemachangerccl_test_/schemachangerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/schemachanger/testdata/end_to_end/alter_table_multiple_commands/alter_table_multiple_commands.definition:5:
        stage-exec [2 args]
        INSERT INTO t(j,i) VALUES($stageKey, $stageKey);
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/schemachangerccl_test_/schemachangerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/schemachanger/testdata/end_to_end/alter_table_multiple_commands/alter_table_multiple_commands.definition:11:
        stage-query [2 args]
        SELECT count(*)=$successfulStageCount FROM t;
        ----
        true
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_15_of_15/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_15_of_15/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_2_of_15/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_11_of_15/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_12_of_15/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_15_of_15
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_15_of_15/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_1_of_15/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_3_of_15/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_7_of_15
    backup.go:105: skipping due to randomness
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_9_of_15
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_12_of_15/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_14_of_15
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_9_of_15/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_1_of_15
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_8_of_15/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_non_revertible_stage_1_of_4/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_12_of_15
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_4_of_15/restore_all_tables_in_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_8_of_15/restore_database
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_stage_1_of_15/restore_database_with_schema_only
=== RUN   TestBackupSuccessMixedVersion_base_alter_table_multiple_commands/post_commit_non_revertible_stage_2_of_4
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_base_alter_table_multiple_commands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/schemachangerccl/BUILD.bazel
+++ b/pkg/ccl/schemachangerccl/BUILD.bazel
@@ -27,7 +27,7 @@ go_test(
         "//pkg/sql/schemachanger:end_to_end_testdata",
     ],
     embed = [":schemachangerccl"],
-    exec_properties = {"test.Pool": "large"},
+    exec_properties = {"test.Pool": "heavy"},
     shard_count = 48,
     tags = ["cpu:3"],
     deps = [

